cpp/thirdparty/CMakeLists.txt

```diff
@@ -51,10 +53,75 @@ endif()
 
 # ## BOOST
 if(MEMILIO_USE_BUNDLED_BOOST)
-    include(BuildBoost)
+    # Ensure that the boost atomic library is used instead of the standard atomic library, where some functionality is only available as of C++20.
+    add_compile_definitions(BOOST_FILESYSTEM_NO_CXX20_ATOMIC_REF)

```

Comment:
can this be added as a `PUBLIC` compile definition of the boost target using `target_compile_definition`? As a rule, it is cleaner set all properties per target instead of globally, and all dependent targets inherit them automatically.